### PR TITLE
[2835] Add location filtering

### PR DESCRIPTION
**Pull request by faissalMT**
### Context

To support location filtering on Find the we need to add it to the API

### Changes proposed in this pull request

- Change to geokit-rails
- Add the filter (latitude+longitude+radius) to the v3 course index API

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by dankmitchell** [comment]
[[L] location filter: Add filtering to api](https://trello.com/c/UkVHbumo/2835-l-location-filter-add-filtering-to-api)

File: config/initializers/geokit_config.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
+# These defaults are used in Geokit::Mappable.distance_to and acts_as_mappable
+Geokit::default_units = :miles # others :kms, :nms, :meters
```
Comment (TobyRet) [inline config/initializers/geokit_config.rb]: @faissalMT - is the Redis cache enabled for this as was for the `geocoder` gem?

Comment (TobyRet) [reply config/initializers/geokit_config.rb]: Just spoken to @misaka about this and perhaps not so important to have the caching enabled now that the geocoding backfilling has been done

File: app/models/course.rb [hunk 2]
```diff
@@ -157,6 +157,10 @@ def new_draft_attributes
     end
   end
 
+  scope :within, ->(range, origin:) do
+    joins(:sites).merge(Site.within(range, origin: origin))
+  end
```
Comment (misaka) [inline app/models/course.rb]: I thought I was able to do `acts_as_mappable` here.

**Pull request merged**